src/Validation/Requires.cs

```diff
@@ -457,6 +457,79 @@ public static void That([DoesNotReturnIf(false)] bool condition, string? paramet
             }
         }
 
+        /// <summary>
+        /// Throws an exception if the specified parameter's value is not null
+        /// <em>and</em> has an element which does not match the given predicate.
+        /// </summary>
+        /// <typeparam name="T">The type of the elements in the sequence.</typeparam>
+        /// <param name="values">The value of the argument.</param>
+        /// <param name="predicate">The predicate used to test the elements.</param>
+        /// <param name="parameterName">The name of the parameter to include in any thrown exception.</param>
+        /// <param name="message">A message to be used in the resulting exception.</param>
+        /// <exception cref="ArgumentException">Thrown if the tested condition is false.</exception>
+        /// <exception cref="ArgumentNullException">Thrown if <paramref name="predicate"/> is <see langword="null"/>.</exception>
```

Comment:
Apparently now also thrown if `values is null`.

Reply:
I'll update the comments.